pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -82,14 +86,26 @@ type streamIngestionProcessor struct {
 	// rewriteToDiffKey Indicates whether we are rekeying a key into a different key.
 	rewriteToDiffKey bool
 
-	// curBatch temporarily batches MVCC Keys so they can be
+	// curKVBatch temporarily batches MVCC Keys so they can be
 	// sorted before ingestion.
 	// TODO: This doesn't yet use a buffering adder since the current
 	// implementation is specific to ingesting KV pairs without timestamps rather
 	// than MVCCKeys.
-	curBatch mvccKeyValues
-	// batcher is used to flush SSTs to the storage layer.
-	batcher           *bulk.SSTBatcher
+	curKVBatch mvccKeyValues
+	// curRangeKVBatch temporarily batches MVCCRangeKeys to be ingested separately
+	// with 'rangeKeySSTWriter' below.
+	curRangeKVBatch mvccRangeKeyValues
+	// batcher is used to flush KVs into SST to the storage layer.
+	batcher *bulk.SSTBatcher
+	// rangeKeySSTWriter is used to flush MVCC range keys into SST to the storage layer.
+	// TODO(casper): extend SSTBatcher to support adding MVCCRangeKeyValue and this
+	// can be replaced.
+	db *kv.DB
+	// In-memory SST file for flushing MVCC range keys
+	rangeKeySstFile *storage.MemFile
+	// Separated SSTWriter to write MVCC range keys and will be replaced after
+	// SSTBatcher supports adding MVCCRangeKeys.
+	rangeKeySSTWriter storage.SSTWriter
```

Comment:
would be nice to split those range related fields into separate struct
rangeState struct {
   ....
}
or similar.
Also would be nice to have a pretty substantial comment the mechanics of how that data should be flushed; frontier updates, etc.